### PR TITLE
mbedOS update

### PR DESCRIPTION
**Pull request by kotylamichal**
Update to work with mbedOS 6.16.0

### CODE REVIEW

File: Applications/IoTBlock-test/main.cpp [hunk 1]
```diff
@@ -13,15 +13,14 @@ using namespace vitroio::sdk;
 #define SERIAL_BAUDRATE 115200
 
 /**
- * Watchdog
+ * Watchdog ticker
```
Comment (Silentsky0) [inline Applications/IoTBlock-test/main.cpp:16]: Do you mean `ticker` or `kicker`?

Comment (kotylamichal) [reply Applications/IoTBlock-test/main.cpp:16]: @Silentsky0 in this case it is ticker which is used to clock watchdog kicker

Comment (Silentsky0) [reply Applications/IoTBlock-test/main.cpp:16]: Alright, resolving

File: docker_build_image.sh [hunk 2]
```diff
@@ -30,7 +30,7 @@ _run_docker(){
     
     if _command_exists docker
     then
-        docker run --rm -it -v $PWD:/home/build vitrotech/docker-mbed-cli /bin/bash -c ./$build_script
+        docker run -i -t --mount=type=bind,source="$(pwd)",destination=/var/mbed -w /var/mbed ghcr.io/armmbed/mbed-os-env /bin/bash -c ./$build_script
```
Comment (Silentsky0) [inline docker_build_image.sh:33]: `--rm` option should be added here as well

**Pull request merged**